services/batch-service.js

```diff
@@ -172,33 +178,13 @@ class BatchService {
         return (batches && batches.length > 0)? batches[0].batchID : null;
     }
 }
-const listBatchConditions = (userID, collaboratorUserIDs, userRole, aUserOrganization, submissionID, userDataCommonsNames) => {
-    const submissionJoin = [
-        {"$lookup": {
-            from: SUBMISSIONS_COLLECTION,
-            localField: "submissionID",
-            foreignField: "_id",
-            as: "batch"
-        }},
-        {"$unwind": {
-            path: "$batch",
-        }}
-    ];
 
-    const validStatusAndSubmissionID = {"submissionID": submissionID, "batch.status": {$in: [NEW, IN_PROGRESS, SUBMITTED, RELEASED, COMPLETED, ARCHIVED, CANCELED, REJECTED, WITHDRAWN]}};
+const isValidBatchRole = (userInfo, aSubmission, collaboratorUserIDs) => {
     const listAllSubmissionRoles = [USER.ROLES.ADMIN, USER.ROLES.FEDERAL_LEAD];
-    if (listAllSubmissionRoles.includes(userRole) || collaboratorUserIDs.length > 0) {
-        return [...submissionJoin, {"$match": {...validStatusAndSubmissionID}}];
-    }
-
-    if (userRole === USER.ROLES.SUBMITTER || userRole === USER.ROLES.USER) {
-        return [...submissionJoin, {"$match": {...validStatusAndSubmissionID, "batch.submitterID": userID}}];
-    }
-
-    if (userRole === USER.ROLES.DATA_COMMONS_PERSONNEL && userDataCommonsNames?.length > 0) {
-        return [...submissionJoin, {"$match": {...validStatusAndSubmissionID, "batch.dataCommons": {$in: userDataCommonsNames}}}];
-    }
-    throw new Error(ERROR.INVALID_SUBMISSION_PERMISSION);
+    const isAll = aSubmission && listAllSubmissionRoles.includes(userInfo?.role) || collaboratorUserIDs?.length > 0;
```

Comment:
1. These permission checks should be in listBatches function in submission.js. 
2. These permission checks don't look like PBAC, more like old role based permission control.

Reply:
1. The permission checks are removed in batch.js.
2. The old role check has been removed. It already exists in submission.js.